owncloudComLibrary/src/test/java/com/owncloud/android/lib/ShareeResponseTest.kt

```diff
@@ -0,0 +1,145 @@
+package com.owncloud.android.lib
+
+import com.owncloud.android.lib.resources.CommonOcsResponse
+import com.owncloud.android.lib.resources.shares.responses.ShareeOcsResponse
+import com.squareup.moshi.JsonAdapter
+import com.squareup.moshi.Moshi
+import com.squareup.moshi.Types
+import junit.framework.Assert.assertEquals
+import junit.framework.Assert.assertTrue
+import org.junit.Assert.assertNotEquals
+import org.junit.Before
+import org.junit.Test
+import java.lang.reflect.Type
+
+class ShareeResponseTest {
+
+    var response: CommonOcsResponse<ShareeOcsResponse>? = null
+
+    @Before
+    fun prepare() {
+        val moshi = Moshi.Builder().build()
+        val type: Type = Types.newParameterizedType(CommonOcsResponse::class.java, ShareeOcsResponse::class.java)
+        val adapter: JsonAdapter<CommonOcsResponse<ShareeOcsResponse>> = moshi.adapter(type)
+        response = adapter.fromJson(EXAMPLE_RESPONSE)
+    }
+
+    @Test
+    fun `check structure - ok - contains meta`() {
+        assertEquals("OK", response?.ocs?.meta?.message!!)
+        assertEquals(200, response?.ocs?.meta?.statusCode!!)
+        assertEquals("ok", response?.ocs?.meta?.status!!)
+        assertTrue(response?.ocs?.meta?.itemsPerPage?.isEmpty()!!)
+        assertTrue(response?.ocs?.meta?.totalItems?.isEmpty()!!)
+    }
+
+    @Test
+    fun `check structure - ok - contains exact`() {
+        assertNotEquals(null, response?.ocs?.data?.exact)
+    }
+
+    @Test
+    fun `check structure - ok - contains groups`() {
+        assertNotEquals(null, response?.ocs?.data?.groups)
+    }
+
+    @Test
+    fun `check structure - ok - contains remotes`() {
+        assertNotEquals(null, response?.ocs?.data?.remotes)
+    }
+
+    @Test
+    fun `check structure - ok - contains users`() {
+        assertNotEquals(null, response?.ocs?.data?.users)
+    }
+
+    @Test
+    fun `check structure - ok - groups contains two items`() {
+        assertEquals(2, response?.ocs?.data?.groups?.size)
+    }
+
+    @Test
+    fun `check structure - ok - users contains two items`() {
+        assertEquals(2, response?.ocs?.data?.users?.size)
+    }
+
+    @Test
+    fun `check structure - ok - exact_users contains one item`() {
+        assertEquals(1, response?.ocs?.data?.exact?.users?.size)
+    }
+
+    @Test
+    fun `check structure - ok - user1 contains additional data`() {
+        assertEquals("user1@user1.com", response?.ocs?.data?.users?.get(0)?.value?.additionalInfo)
+    }
+
+    @Test
+    fun `check structure - ok - user2 does not contain additional data`() {
+        assertEquals(null, response!!.ocs.data.users!![1].value!!.additionalInfo)
+    }
+
+    companion object {
+        val EXAMPLE_RESPONSE = """
+{
+    "ocs": {
+        "data": {
+            "exact": {
+                "groups": [],
```

Comment:
I like it, but i think that we could create a new responses package with json files and import those files in the test. That way we will have all responses in the same package as real json and not in companion as strings. What do you think? I could help to do it

Reply:
Sounds good

Reply:
Ok, tell me if you need some help to do it